test/index.js

```diff
@@ -1 +1,138 @@
+                        '<h1 class="bigtitle">My text</h1>');
+    eq(H.attr('class', node), just('bigtitle'));
+  });
+  it('returns value of attribute for given node #2', () => {
+    const node = S.pipe([H.parse,
+                         R.nth(0)],
+                        '<p id="main-paragraph">What a great webpage!</p>');
+    eq(H.attr('id', node), just('main-paragraph'));
+  });
+  it('returns value of arbitrary attribute', () => {
+    const node = S.pipe([H.parse,
+                         R.nth(0)],
+                        '<p myattribute="booyah">What a great webpage!</p>');
+    eq(H.attr('myattribute', node), just('booyah'));
+  });
+  it('returns Nothing if attribute does not exist', () => {
+    const node = S.pipe([H.parse,
+                         R.nth(0)],
+                        '<p id="main-paragraph">What a great webpage!</p>');
+    eq(H.attr('style', node), nothing());
+  });
+  it('returns empty string value for boolean attributes', () => {
+    const node = S.pipe([H.parse,
+                         R.nth(0)],
+                        '<input type="checkbox" checked>');
+    eq(H.attr('checked', node), just(''));
+  });
+});
+
+describe('children', () => {
+  it('returns child nodes given parent node #1', () => {
+    const parent = S.pipe([H.parse,
+                           R.filter(H.is('html')),
+                           R.nth(0)],
+                          GOOD_HTML);
+    // Assert children html
+    eq(S.pipe([H.children, R.map(H.html), R.join('')], parent),
+       fs.readFileSync(__dirname + '/children.html', 'utf8'));
+    // Assert children object
+    eq(S.pipe([H.children, R.map(stripNodes)], parent),
+       require('./children.json'));
+  });
+});
+
+describe('find', () => {});
+
+describe('html', () => {});
+
+describe('is', () => {});
+
+describe('next', () => {});
+
+describe('Node', () => {
+  it('returns Node given htmlparser2 _node', () => {
+    const _node = {
+      name: '!doctype',
+      data: '!DOCTYPE html',
+      type: 'directive',
+      next: null,
+      prev: null,
+      parent: null
+    };
+    const node = H.Node(_node);
+    eq(node.value, _node);
+    eq(node['_@@type'], 'sanctuary-html/Node');
+    eq(R.toString(node), 'Node("<!DOCTYPE html>")');
+    eq(R.equals(node, H.Node(_node)), true);
+  });
+
+  // Test equals metods
+});
+
+describe('parent', () => {});
+
+describe('parse', () => {
+  it('parse well formed html', () => {
+    // Assert non-circular htmlparser2 object
+    const simple = '<html></html>';
+    const simpleNodes = H.parse(simple);
+    const simpleExpected = {
+      'attribs': {},
+      'children': [],
+      'name': 'html',
+      'next': null,
+      'parent': null,
+      'prev': null,
+      'type': 'tag',
+    };
+    eq(simpleNodes, [H.Node(simpleExpected)]);
+    // Assert circular htmlparser2 object after stripping circular values.
+    const circularNodes = H.parse(GOOD_HTML);
+    eq(R.map(stripNodes, circularNodes), GOOD_OBJ);
```

Comment:
Great idea to strip circular references @davidchambers, much happier with the second assertion here in place.